### PR TITLE
Add OpenHarmony targets

### PR DESCRIPTION
**Pull request by Amanieu**
- `aarch64-unknown-linux-ohos`
- `armv7-unknown-linux-ohos`

Compiler team MCP: https://github.com/rust-lang/compiler-team/issues/568

### CODE REVIEW

**Comment by workingjubilee** [comment]
I hate to be a pedantic bore about minor points of protocol that mostly don't really matter, but I was cursed at birth with an editor's eyes:

Recently, `{arch}-fuchsia` was changed to `{arch}-unknown-fuchsia` for Toolchain Reasons. Are we sure we want to skip the vendor "field" in this tuple?

**Comment by ehuss** [comment]
r? compiler

File: src/doc/rustc/src/platform-support/openharmony.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
+# `target-name-here`
```
Comment (ehuss) [inline src/doc/rustc/src/platform-support/openharmony.md:1]: Please don't forget to also update https://github.com/rust-lang/rust/blob/HEAD/src/doc/rustc/src/platform-support.md and add this file to https://github.com/rust-lang/rust/blob/master/src/doc/rustc/src/SUMMARY.md.

**Comment by Amanieu** [comment]
> Recently, `{arch}-fuchsia` was changed to `{arch}-unknown-fuchsia` for Toolchain Reasons. Are we sure we want to skip the vendor "field" in this tuple?

I really don't feel too strongly about it. The C toolchain uses `aarch64-linux-ohos` but then again so do many toolchains for which we use `-unknown`. In the end it's just a naming thing and doesn't really matter.

**Comment by workingjubilee** [comment]
Yeah, I don't _really_ care either, my only preference is against name churn. So I just wanted to ask if anyone had doublechecked that, as "C toolchain" might mean "Clang", and I don't know if the same rationales as in https://github.com/rust-lang/rust/pull/106429 will or will not apply because I'm just another ignorant 外国人 who only heard of OpenHarmony today.

File: src/doc/rustc/src/platform-support/openharmony.md (hunk 1 above)
Comment (Amanieu) [reply src/doc/rustc/src/platform-support/openharmony.md]: Done.

**Comment by tmandry** [comment]
To add to what @djkoloski said, we consider `x86_64-unknown-fuchsia` to be the "canonical" triple for both clang and now rustc. Consistency with other targets and across compilers (or frontend/backend) is desirable.

If rustc also supported dropping vendor info for any target where it's `unknown` that would be fine, but you would have to decide if you want to handle that in rustup and possibly other places. For now we just went with what's most consistent.

**Comment by petrochenkov** [comment]
r=me after squashing commits.
@rustbot author

**Pull request merged**